src/ansys/fluent/core/launcher/launcher.py

```diff
@@ -764,162 +982,11 @@ def launch_fluent(
-                    )
-                else:
-                    raise RuntimeError(
-                        "Case and data file cannot be read in meshing mode."
-                    )
-
-            return session
-        except Exception as ex:
-            logger.error(f"Exception caught - {type(ex).__name__}: {ex}")
-            raise LaunchFluentError(launch_cmd) from ex
-        finally:
-            server_info_file = Path(server_info_file_name)
-            if server_info_file.exists():
-                server_info_file.unlink()
-    elif fluent_launch_mode == LaunchMode.PIM:
-        logger.info(
-            "Starting Fluent remotely. The startup configuration will be ignored."
-        )
-
-        if product_version:
-            fluent_product_version = "".join(product_version.split("."))[:-1]
-        else:
-            fluent_product_version = None
-
-        return launch_remote_fluent(
-            session_cls=new_session,
-            start_transcript=start_transcript,
-            product_version=fluent_product_version,
-            cleanup_on_exit=cleanup_on_exit,
-            meshing_mode=meshing_mode,
-            dimensionality=version,
-            launcher_args=argvals,
-        )
-
-    elif fluent_launch_mode == LaunchMode.CONTAINER:
-        args = _build_fluent_launch_args_string(**argvals).split()
-        if meshing_mode:
-            args.append(" -meshing")
-
-        if dry_run:
-            if container_dict is None:
-                container_dict = {}
-            config_dict, *_ = configure_container_dict(args, **container_dict)
-            from pprint import pprint
-
-            print("\nDocker container run configuration:\n")
-            print("config_dict = ")
-            if os.getenv("PYFLUENT_HIDE_LOG_SECRETS") != "1":
-                pprint(config_dict)
-            else:
-                config_dict_h = config_dict.copy()
-                config_dict_h.pop("environment")
-                pprint(config_dict_h)
-                del config_dict_h
-            return config_dict
-
-        port, password = start_fluent_container(args, container_dict)
-
-        session = new_session(
-            fluent_connection=FluentConnection(
-                port=port,
-                password=password,
-                cleanup_on_exit=cleanup_on_exit,
-                start_transcript=start_transcript,
-                launcher_args=argvals,
-                inside_container=True,
-            ),
-            remote_file_handler=RemoteFileHandler(),
-        )
-
-        if start_watchdog is None and cleanup_on_exit:
-            start_watchdog = True
-        if start_watchdog:
-            logger.debug("Launching Watchdog for Fluent container...")
-            watchdog.launch(os.getpid(), port, password)
-
-        return session
+    launcher = create_launcher(argvals.pop("fluent_launch_mode"))
+    return launcher.launch(**argvals)
```

Comment:
@hpohekar AFAICS, this looks very good. 

One thing I'm wondering is whether it might also be useful to actually _construct_ the launcher object (of the given type) with the launch arguments as its state. Then that object can be passed around, reused etc. And then when it comes to launch, the caller also choose to override (or not override) any given arguments. The launcher interface should perhaps simply be a callable (via `__call__`).

Reply:
@seanpearsonuk Yes, I agree. We will implement this once we resolved current failing checks. Thank you.

Reply:
Done. Thank you.